yarn-project/blob-lib/src/blob.ts

```
@@ -68,16 +77,14 @@ export class Blob {
    * @returns A Blob created from the array of fields.
    */
   static async fromFields(fields: Fr[], multiBlobFieldsHash?: Fr): Promise<Blob> {
-    if (fields.length > FIELD_ELEMENTS_PER_BLOB) {
-      throw new Error(
-        `Attempted to overfill blob with ${fields.length} elements. The maximum is ${FIELD_ELEMENTS_PER_BLOB}`,
-      );
+    if (fields.length > FIELDS_PER_BLOB) {
+      throw new Error(`Attempted to overfill blob with ${fields.length} elements. The maximum is ${FIELDS_PER_BLOB}`);
     }
 
     const data = Buffer.concat([serializeToBuffer(fields)], BYTES_PER_BLOB);
 
     // This matches the output of SpongeBlob.squeeze() in the blob circuit
-    const fieldsHash = multiBlobFieldsHash ? multiBlobFieldsHash : await poseidon2Hash(fields);
+    const fieldsHash = multiBlobFieldsHash ? multiBlobFieldsHash : await Blob.getFieldsHash(fields);
```

I'd push back against making `multiBlobFieldsHash` optional. It's likely we forget setting it, and if we have multiple blobs for a block, it will default to an incorrect value.